pubsub/drivertest/drivertest.go

```diff
@@ -374,15 +377,32 @@ func testCancelSendReceive(t *testing.T, newHarness HarnessMaker) {
 	}
 	defer cleanup()
 
+	canceledCtx, cancel := context.WithCancel(ctx)
 	cancel()
 
 	m := &pubsub.Message{}
-	if err := top.Send(ctx, m); !isCanceled(err) {
+	if err := top.Send(canceledCtx, m); !isCanceled(err) {
 		t.Errorf("top.Send returned %v (%T), want context.Canceled", err, err)
 	}
-	if _, err := sub.Receive(ctx); !isCanceled(err) {
+	if _, err := sub.Receive(canceledCtx); !isCanceled(err) {
 		t.Errorf("sub.Receive returned %v (%T), want context.Canceled", err, err)
 	}
+
+	// Repeat with a sub.Receive that's interrupted.
+	delayedCancelCtx, cancel := context.WithCancel(ctx)
+	ch := make(chan struct{})
+	go func() {
+		if _, err := sub.Receive(delayedCancelCtx); !isCanceled(err) {
+			t.Errorf("sub.Receive returned %v (%T), want context.Canceled", err, err)
+		}
+		close(ch)
+	}()
+	// Give the goroutine some time to spin up and be blocking in Receive.
+	// There's no guarantee it will get there, but tests will at least be flaky
+	// if it gets there sometimes.
+	time.Sleep(50 * time.Millisecond)
+	cancel()
+	<-ch
```

Comment:
Add a `<-time.After(10*time.Second)` here. If Receive ignores cancelation, then this test will fail by timing out, which will take 10 minutes if you don't make it shorter.

Reply:
Done.